web/src/app/services/selected-term.service.ts

```diff
@@ -0,0 +1,106 @@
+import {Injectable} from '@angular/core';
+import {Term} from 'yacs-api-client';
+import {Subject, Subscription} from 'rxjs';
+
+@Injectable()
+export class SelectedTermService {
+    // a subject that can be written to and subscribed to for the selected term
+    public selectedTerm$: Subject<Term>;
+    // hold the terms in two forms: a mapping between the uuid and the term, and an array for ordering
+    protected terms: Map<string, Term>;
+    protected termsOrdinal: Term[];
+    // cache the current ordinal and uuid
+    protected _currentOrdinal: number;
+    protected _currentUUID: string;
+    protected _currentId: string;
+
+    constructor() {
+        this.terms = new Map<string, Term>();
+        this.termsOrdinal = new Array<Term>();
+        this.selectedTerm$ = new Subject<Term>();
+        this._currentOrdinal = 0;
+        // acquire all the terms upon load
+        Term.all().then(terms => {
+            terms.data.forEach(term => {
+                // place the terms in the map and the array
+                this.terms.set(term.uuid, term);
+                this.termsOrdinal.push(term);
+            });
+            if (localStorage['atFirstTerm'] === 'true') {
+                // if the local storage term was the first term (the most recent)
+                // set the selected term to the most recent instead of restoring
+                // the last setting, which may unnecessarily obfuscate the most
+                // recent term
+                this.setSelectedTermByOrdinal(0);
+            } else {
+                // otherwise restore the selectedTerm from localStorage, verifying its validity
+                // and defaulting to the most recent
+                if (localStorage['selectedTerm'] !== undefined) {
+                    const localTerm = this.terms.get(localStorage['selectedTerm']);
+                    if (localTerm === undefined) {
+                        this.setSelectedTermByOrdinal(0);
+                    } else {
+                        this._currentUUID = localTerm.uuid;
```

Comment:
IIRC, I did this because the term id may reference a different term. This is particularly noticeable in the case of stored preferences, but on the edge case that a new term becomes available while a user is using the service it may pose an issue. It's low impact, so we can consolidate this.

Reply:
I don't understand. Both ids and uuids are guaranteed unique.